resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DeletedKubernetesClusterChecker.scala

```diff
@@ -0,0 +1,37 @@
+package com.broadinstitute.dsp
+package resourceValidator
+
+import cats.effect.{Concurrent, Timer}
+import cats.mtl.ApplicativeAsk
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.google2.GKEModels.KubernetesClusterId
+import org.broadinstitute.dsde.workbench.model.TraceId
+
+object DeletedKubernetesClusterChecker {
+  def impl[F[_]: Timer](
+    dbReader: DbReader[F],
+    deps: KubernetesClusterCheckerDeps[F]
+  )(implicit F: Concurrent[F], logger: Logger[F], ev: ApplicativeAsk[F, TraceId]): CheckRunner[F, KubernetesCluster] =
+    new CheckRunner[F, KubernetesCluster] {
+      override def appName: String = resourceValidator.appName
+
+      override def configs = CheckRunnerConfigs(s"deleted-kubernetes-clusters", shouldAlert = true)
+
+      override def dependencies: CheckRunnerDeps[F] = deps.checkRunnerDeps
+
+      override def checkResource(cluster: KubernetesCluster, isDryRun: Boolean)(
+        implicit ev: ApplicativeAsk[F, TraceId]
+      ): F[Option[KubernetesCluster]] = checkKubernetesClusterStatus(cluster, isDryRun)
+
+      override def resourceToScan: fs2.Stream[F, KubernetesCluster] = dbReader.getDeletedAndErroredKubernetesClusters
+
+      def checkKubernetesClusterStatus(cluster: KubernetesCluster, isDryRun: Boolean)(
+        implicit ev: ApplicativeAsk[F, TraceId]
+      ): F[Option[KubernetesCluster]] =
+        for {
+          clusterOpt <- deps.gkeService.getCluster(
+            KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
```

Comment:
you need to delete the cluster here (also check `isDryRun`)

Reply:
was fixing a compilation error first, just pushed the changes to delete the nodepool and cluster and to check `isDryRun` :)